### ISSUE TITLE
Can we define the layout to be used by pdftotext while converting the pdf to text file?

### ISSUE THREAD

**Issue by gtambi143** [invoice-x/invoice2data#205]

**Comment by m3nu** [comment]
You can add a new argument to the pdf2text wrapper [here](https://github.com/invoice-x/invoice2data/blob/master/src/invoice2data/input/pdftotext.py).

**Comment by sudeepjd** [comment]
I found this to be a problem for me as well pdftotext conversion currently hardcodes -layout as the conversion method, in some cases, I find -table to be better suited for the conversion... 

I can submit a PR to add in a layout parameter which can then get passed to the to_text functions in each of the .input modules... Should I?

**Comment by m3nu** [comment]
Yes. Always useful to have it as option. Just keep the current setting as default.

**Comment by rmilecki** [comment]
The problem I see is that poppler's `pdftotext` doesn't support `-simple` or `-table`. Support for those layouts was developed in Xpdf after it has been forked by the poppler project.

Today most Linux distributions switched to `poppler` for providing `pdftotext` and similar tools. Ideally support for those extra layouts should be ported from Xpdf to poppler but that seems like a huge task. I created request issue in poppler project to see if there is any idea/interest around that: https://gitlab.freedesktop.org/poppler/poppler/-/issues/1419

**Comment by bosd** [comment]
Resolved in **1.0.0**. The default input-backend cascade is now `pdfium → pdftotext` (invocation of pdftotext uses `-layout`, which is the multi-column-aware mode). A template can pin its authored backend explicitly:

```yaml
input_module: pdftotext   # or 'pdfium', 'pdfminer_six', ...
```

For your `-simple` (single-column) use case, either:

- use `input_module: pdfium` (pypdfium2, no layout mode — closest to `-simple`), or
- author your regex against the `-layout` output (what most bundled templates do) and use `extract._regex` (which is `re`-compatible) with `\s+` to handle the spacing.

Full backend list + benchmarks: [docs/backend-benchmark.md](https://github.com/invoice-x/invoice2data/blob/master/docs/backend-benchmark.md).

Closing as done.
